File: rock-paper-scissor.pl
#!/usr/bin/perl

use strict;
use warnings;

my @choices = qw(rock paper scissors);
my $player_score = 0;
my $computer_score = 0;

print "Welcome to Rock-Paper-Scissors game!\n";

while (1) {
    print "Choose your move (rock, paper, or scissors): ";
    chomp(my $player_choice = lc <STDIN>);
    my $computer_choice = $choices[int rand @choices];
    
    # Determine winner
    my $result;
    if ($player_choice eq $computer_choice) {
        $result = "tie";
    } elsif ($player_choice eq "rock" && $computer_choice eq "scissors" ||
             $player_choice eq "paper" && $computer_choice eq "rock" ||
             $player_choice eq "scissors" && $computer_choice eq "paper") {
        $result = "player";
        $player_score++;
    } else {
        $result = "computer";
        $computer_score++;
    }
    
    # Print result
    print "You chose $player_choice, and the computer chose $computer_choice. ";
    if ($result eq "tie") {
        print "It's a tie!\n";
    } else {
        print "The $result wins this round.\n";
    }
    
    # Print scores
    print "Current scores: You $player_score - $computer_score Computer\n";
    
    # Ask to play again
    print "Do you want to play again? (y/n): ";
    chomp(my $play_again = lc <STDIN>);
    last if $play_again ne "y";
}

print "Thanks for playing!\n";

File: rock-paper-scissor.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strings"
	"time"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	// Set up game
	rand.Seed(time.Now().UnixNano())
	choices := []string{"rock", "paper", "scissors"}
	wins := [][]int{
		{0, 1, -1},
		{-1, 0, 1},
		{1, -1, 0},
	}

	// Play game
	fmt.Println("Let's play Rock-Paper-Scissors!")
	for {
		// Get player choice
		fmt.Print("Enter your choice (rock/paper/scissors): ")
		playerChoice, err := reader.ReadString('\n')
		if err != nil {
			fmt.Println("Error reading input:", err)
			continue
		}
		playerChoice = strings.TrimSpace(playerChoice)

		// Check player choice is valid
		playerIndex := -1
		for i, choice := range choices {
			if choice == playerChoice {
				playerIndex = i
				break
			}
		}
		if playerIndex == -1 {
			fmt.Println("Invalid choice, please enter rock, paper, or scissors.")
			continue
		}

		// Get computer choice
		computerIndex := rand.Intn(3)
		fmt.Println("Computer chose", choices[computerIndex])

		// Determine winner
		result := wins[playerIndex][computerIndex]
		switch result {
		case 1:
			fmt.Println("You win!")
		case 0:
			fmt.Println("It's a tie!")
		case -1:
			fmt.Println("Computer wins!")
		}

		// Ask if player wants to play again
		fmt.Print("Play again? (y/n): ")
		playAgain, err := reader.ReadString('\n')
		if err != nil {
			fmt.Println("Error reading input:", err)
			continue
		}
		playAgain = strings.TrimSpace(playAgain)
		if playAgain != "y" {
			break
		}
	}
}
